include/nbl/builtin/hlsl/sampling/quotient_and_pdf.hlsl

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2018-2025 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_SAMPLING_QUOTIENT_AND_PDF_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SAMPLING_QUOTIENT_AND_PDF_INCLUDED_
+
+#include "nbl/builtin/hlsl/concepts/vector.hlsl"
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace sampling
+{
+
+template<typename T, typename F>
+NBL_BOOL_CONCEPT spectral_of = concepts::Vectorial<T> || concepts::Scalar<T> || concepts::Scalar<F>;
```

Comment:
thats a really weird concept name, why are we checking two types at the same time, we could just put it in the `NBL_PRIMARY_REQUIRES` below

Reply:
I'll have to change it a bit, but it's to check the type is fine for `spectral_type` so a combination of vectorial or scalar and is floating point(?).